Delete OpenShift 3.x identity provider

stianst opened the pull request:
Closes #34330

Signed-off-by: stianst <stianst@gmail.com>

stianst committed:
Delete OpenShift 3.x identity provider

Closes #34330

Signed-off-by: stianst <stianst@gmail.com>

jonkoops commented on js/apps/admin-ui/cypress/e2e/identity_providers_test.spec.ts:
```diff
@@ -88,11 +88,6 @@ describe("Identity provider test", () => {
-        },
         {
           testName: "Openshift-v4",
           displayName: "Openshift v4",
```
Perhaps we also re-name labels with 'Openshift v4' to only 'Openshift'?

stianst replied:
Probably wouldn't do that since it would be a potential breaking change, and I'd say it doesn't hurt to just leave it as is.

jonkoops reviewed (dismissed):
LGTM, but would this not be a breaking change in a minor release? Is that acceptable with our new versioning scheme? Should we not document this in the changelog?

mabartos commented:
@stianst 

1. I think also the https://github.com/keycloak/keycloak/blob/main/docs/documentation/server_admin/topics/identity-broker/social/openshift.adoc#openshift-3 doc should be updated.
2. As @jonkoops pointed out, I'd also vote to include some paragraph in the release notes.

_just thought: It's unfortunate that we haven't managed any deprecation period for it, but as it has reached the EOL, there's nothing we can do, I guess._

stianst commented:
> LGTM, but would this not be a breaking change in a minor release? Is that acceptable with our new versioning scheme? Should we not document this in the changelog?

No, we can remove support for things that are no longer supported. OpenShift v3 is no longer even supported as an ELS.

stianst commented:
> @stianst
> 
> 1. I think also the https://github.com/keycloak/keycloak/blob/main/docs/documentation/server_admin/topics/identity-broker/social/openshift.adoc#openshift-3 doc should be updated.
> 2. As @jonkoops pointed out, I'd also vote to include some paragraph in the release notes.
> 
> _just thought: It's unfortunate that we haven't managed any deprecation period for it, but as it has reached the EOL, there's nothing we can do, I guess._

Not really, as we're not the once responsible for EOL of OpenShift v3.

stianst commented:
@mabartos @jonkoops I've removed the section from the docs and also added a mention in the release notes. As OCP v3 was EOL in June 2024 I don't consider this a breaking change, or something that should have had a deprecation period on our end (although maybe we could have deprecated it once OCP 3 was moved to ELS).

mabartos reviewed (approved):
LGTM
